portfolio/src/main/java/com/google/sps/servlets/LogInServlet.java

```diff
@@ -24,11 +24,20 @@
 @WebServlet("/login")
 public class LogInServlet extends HttpServlet {
   
+  final String ROOT_URL = "/";
+  
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("application/json;");
     UserService userService = UserServiceFactory.getUserService();
-    boolean loggedIn = userService.isUserLoggedIn();
-    response.getWriter().println(new Gson().toJson(loggedIn));
+    List<String> loginInfo = new ArrayList<String>();
```

Comment:
JSON has an efficient representation of bool, which you should use instead of strings (so this should be List<Boolean>). This will also make your JavaScript cleaner.

Reply:
Also, why is this a list? It doesn't seem like it will ever have more than one element, so an object with a single, named boolean field is more appropriate.

Reply:
Can you have a list of multiple types in java though? The other thing I'm sending is the login or logout url respectively.